monitoring-dashboard/components/org.wso2.ei.dashboard.core/src/main/java/org/wso2/ei/dashboard/core/data/manager/InMemoryDataManager.java

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright (c) 2022, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ *
+ */
+
+package org.wso2.ei.dashboard.core.data.manager;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.wso2.ei.dashboard.core.exception.DashboardServerException;
+import org.wso2.ei.dashboard.core.rest.delegates.heartbeat.HeartbeatObject;
+import org.wso2.ei.dashboard.core.rest.model.GroupList;
+import org.wso2.ei.dashboard.core.rest.model.NodeList;
+import org.wso2.ei.dashboard.core.rest.model.NodeListInner;
+import java.util.HashMap;
+
+/**
+ * Performs in-memory map operations.
+ */
+public final class InMemoryDataManager implements DataManager {
+
+    private static final Logger logger = LogManager.getLogger(InMemoryDataManager.class);
+    private static final String GROUP_ID = "groupId";
+    private static final String NODE_ID = "nodeId";
+    private static final String SERVICE_INFO = "serviceInfo";
+    public static HashMap<String, HashMap> heartBeatStore;
+    public static HashMap<String, HashMap> serviceInfoStore;
+
+    public InMemoryDataManager() {
+        heartBeatStore = new HashMap<>();
+        serviceInfoStore = new HashMap<>();
+        logger.debug("heartBeatStore and serviceInfoStore created ");
+    }
+
+    @Override
+    public boolean insertHeartbeat(HeartbeatObject heartbeat, String accessToken) {
+        if (heartbeat == null) {
+            logger.debug("Received heartbeat object is null");
+            return false;
+        }
+        try {
+            HashMap map = new HashMap<>();
+            map.put(GROUP_ID, heartbeat.getGroupId());
+            map.put(NODE_ID, heartbeat.getNodeId());
+            map.put("interval", heartbeat.getInterval());
+            map.put("mgtUrl", heartbeat.getMgtApiUrl());
+            map.put("timeStamp", heartbeat.getTimestamp());
+            map.put("accessToken", accessToken);
+            String keyString = heartbeat.getGroupId() + heartbeat.getNodeId();
+            heartBeatStore.put(keyString, map);
+            logger.info("Inserted heartbeat to node " + heartbeat.getNodeId() +  " of group " + heartbeat.getGroupId());
```

Comment:
Shouldn't these be debug logs? @dulanjalidilmi

Reply:
Since insertion occurs at MI node registration, let's keep this as an info log. So that user can see that the particular node has been registered successfully with the dashboard.